agent/consul/state/intention.go

```diff
@@ -792,6 +775,37 @@ func (s *Store) IntentionDecision(
 	return resp, nil
 }
 
+// AuthorizeIntentionTarget determines whether the destination is covered by the given intention
+// and whether the intention action allows a connection.
+func AuthorizeIntentionTarget(target, targetNS string, ixn *structs.Intention, matchType structs.IntentionMatchType) (bool, bool) {
```

Comment:
May want to note that this logic formerly was in `(agent/connect).SpiffeIDService.Authorize`. Is the former method used anymore in mainline consul+envoy code?

Reply:
It was still being used in `Agent.ConnectAuthorize`. I don't really see the need to keep the old Authorize method, nor to keep it in the CertURI interface, so I removed it in favor of the new one.